src/server/pegasus_server_impl.cpp

```diff
@@ -1773,65 +1814,59 @@ ::dsn::error_code pegasus_server_impl::sync_checkpoint()
 
     int64_t last_durable = last_durable_decree();
     int64_t last_commit = last_committed_decree();
-    dassert(last_durable <= last_commit, "%" PRId64 " VS %" PRId64, last_durable, last_commit);
+    dcheck_le_replica(last_durable, last_commit);
```

Comment:
I suggest separating the logging refactoring into a new PR.

Reply:
https://github.com/XiaoMi/pegasus/pull/503